### PR TITLE
slirp: add a new option -e FD

### PR DESCRIPTION
**Pull request by giuseppe**
if specified it tells slirp4netns what FD must be monitored for
events.  On an event on the FD the program exits with success.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: slirp.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef SLIRP_H
+# define SLIRP_H
```
Comment (AkihiroSuda) [inline slirp.h]: nit: indent

**Comment by AkihiroSuda** [comment]
SGTM, but why not just kill?

File: slirp.h (hunk 1 above)
Comment (giuseppe) [reply slirp.h]: I did that on purpose as the define is a level below the #ifndef, but if you don't like to indent the C preprocessor macros, I can drop the extra space

File: main.c [hunk 2]
```diff
@@ -30,10 +31,7 @@ static int nsenter(pid_t target_pid)
 		perror(netns);
 		return netnsfd;
 	}
-	if (setns(usernsfd, CLONE_NEWUSER) < 0) {
-		perror("setns(CLONE_NEWUSER)");
-		return -1;
-	}
+	setns(usernsfd, CLONE_NEWUSER);
```
Comment (AkihiroSuda) [inline main.c]: why?

Comment (giuseppe) [reply main.c]: in podman we already joined the userns, and attempting to join it again will fail.  I think it is fine to ignore errors here, as the join network namespace would still fail later on.  What do you think?

**Comment by giuseppe** [comment]
with these changes I am able to configure a network for podman rootless container.  Cleaning up and opening a PR later on today for podman

File: main.c (hunk 2 above)
Comment (AkihiroSuda) [reply main.c]: I think we should fail here unless already joined, but can be amother pr

**Pull request merged**